docs/sources/flow/reference/config-blocks/import.file.md

```diff
@@ -0,0 +1,73 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/config-blocks/import.file/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/config-blocks/import.file/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/config-blocks/import.file/
+- /docs/grafana-cloud/send-data/agent/flow/reference/config-blocks/import.file/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/config-blocks/import.file/
+description: Learn about the import.file configuration block
+labels:
+  stage: beta
+title: import.file
+---
+
+# import.file
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" version="<AGENT_VERSION>" >}}
+
+The `import.file` block imports custom components from a file and exposes them to the importer. `import.file` blocks must be given a label which determines the namespace where custom components are exposed.
+
+[module]: {{< relref "../../concepts/modules.md" >}}
+
+## Usage
+
+```river
+import.file "NAMESPACE" {
+  filename = FILENAME
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`filename`       | `string`   | Path of the file on disk to watch | | yes
+`detector`       | `string`   | Which file change detector to use (fsnotify, poll) | `"fsnotify"` | no
+`poll_frequency` | `duration` | How often to poll for file changes | `"1m"` | no
+`is_secret`      | `bool`     | Marks the file as containing a [secret][] | `false` | no
```

Comment:
It doesn't make sense to me for `is_secret` to exist on `import.file` and `import.http`; those are only used when exporting values but we're never going to export them here. Can we make sure to remove these options from the code before we ship v0.40?

Reply:
I made the changes import_file : https://github.com/grafana/agent/pull/6350/commits/1221f5d1e40183b71d189b0c311b254ec8394689